src/Core/DevOps/StaticAnalyze/PHPStan/Rules/Internal/InternalClassRule.php

```diff
@@ -31,7 +39,20 @@ class InternalClassRule implements Rule
 
     private const INTERNAL_NAMESPACES = [
         '\\DevOps\\StaticAnalyze',
-        '\\Framework\\Demodata',
+    ];
+    private const SUBSCRIBER_EXCEPTIONS = [
+        RefreshIndexCommand::class,
```

Comment:
this exception is not needed i guess

Reply:
ok, what about the others (JsonFieldSerializer, OrderLineItemEntity->getPromotion() / ->getPromotionId()
DocumentConfigLoader)

Reply:
For:
`OrderLineItemEntity->getPromotion() / ->getPromotionId()`
I don't see why they are internal we should make them public IMHO

`DocumentConfigLoader`
We should split the actual loading and the "caching" and resetting in two services, the actual loading should be public (either by adding a public abstract class or just remove internal from the actual loader class)

`JsonFieldSerializer`
All serializer are internal, this is not public api for us, and i would like to keep it that way, can you explain the use case why you need to access that class? Which cache keys to you need to generate and how?

Reply:
@keulinho `JsonFieldSerializer` - look at your own docs: https://developer.shopware.com/docs/guides/plugins/plugins/framework/store-api/add-caching-for-store-api-route
(generateKey() at the very end)

of course, the current implementation of `encodeJson` is quite trivial and i could just copy that into my code. but that doesn't feel right.

Reply:
@netzperfekt I adjusted some code fragments you mentioned:

- EntityIndexingMessage: __construct is no more `@internal`
- JsonFieldSerializer: I implemented a new Json class which can be used - docs update follows
- OrderLineItemEntity->getPromotion(): Is no more `@internal`
- DocumentConfigLoader - Is now marked as public API, but only the `load` function to allow calling the service.
- AppEntity: is no more `@internal`